torch/csrc/dynamo/compiled_autograd.h

```diff
@@ -432,6 +434,10 @@ class CompiledNodeArgs {
   }
   CompiledNodeArgs(const CompiledNodeArgs&) = delete;
 
+  void set_required_shape(const at::Tensor& tensor, c10::SymIntArrayRef shape) {
+    _compiler.tensor_args.lookup(tensor).required_shape = shape;
+  }
```

Comment:
What are the lifecycle implications of storing a ref? Maybe add a comment?

Reply:
This isn't storing a ref, it is doing a copy.

Reply:
Isn't `c10::SymIntArrayRef` a ref? You are copying the structure, not the contents.

To quote the headerdoc:

```
/// This class does not own the underlying data, it is expected to be used in
/// situations where the data resides in some other buffer, whose lifetime
/// extends past that of the ArrayRef. For this reason, it is not in general
/// safe to store an ArrayRef.
```

Reply:
But `_compiler.tensor_args.lookup(tensor).required_shape` is a vector.
```
vector = ref
``
means a copy.

Reply:
Hmm, as long as the copy semantics of the ref move the data correctly, I agree. I assumed from the headerdoc that the contents don't get copied, only the structure.